### PR TITLE
[refactor] update to new uglify while still supporting old API in Fac…

### PR DESCRIPTION
**Pull request by jcrugzz**
…tory.js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,8 +56,8 @@
     "eslint-plugin-json": "^1.2.0",
     "eslint-plugin-mocha": "^4.11.0",
     "eslint-plugin-react": "^7.1.0",
-    "istanbul": "~0.4.5",
     "mocha": "~3.2.0",
-    "npm": "~3.10.10"
+    "npm": "~3.10.10",
```
Comment (indexzero) [inline package.json]: Do we want to try to update this dep as part of this work?

Comment (jcrugzz) [reply package.json]: yea I can do that. its only used in tests.

File: minifiers/uglifyjs.js [hunk 2]
```diff
@@ -32,25 +36,42 @@ module.exports = function uglify(options, done) {
...
+
+  if (results.error) return done(results.error);
 
   //
   // Get hash for content. The sourceMap and JS content need to be stored under the same hash.
   //
   const fingerprint = fingerprinter(options.file, { content: results.code, map: true });
 
+
+
```
Comment (msluther) [inline minifiers/uglifyjs.js]: nit: extra lines. ;)

File: package.json [hunk 3]
```diff
@@ -55,8 +56,8 @@
     "eslint-plugin-json": "^1.2.0",
     "eslint-plugin-mocha": "^4.11.0",
     "eslint-plugin-react": "^7.1.0",
-    "istanbul": "~0.4.5",
     "mocha": "~3.2.0",
```
Comment (msluther) [inline package.json]: worthwhile to update any of the other devdeps while you're here? mocha, etc.

**Pull request merged**